lib/bowling_score.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+# BowlingScore used to take a single string showing the score of each throw of
+# the game as input and calculate the total score
+class BowlingScore
+  attr_accessor :all_throws
+
+  def initialize(all_throws, full_game = false)
+    @all_throws = all_throws.split('')
```

Comment:
I'd be tempted to try and put in an upper limit to the size of the array you have here. The maximum you have using a normal array (in theory) is 21 because 9 x 2, followed by 1 x 3 for the end game.

I think what you may have to do here is convert to either a 2D array (so your first 9 rounds have a maximum array size of 2, and your last round has a maximum array size of 3). It could get potentially a little bit more complicated (so for example, in the last round, you don't use the last element of the size 3 array, unless a strike or spare has been obtained in previous throws in the round)...but at the moment you will have a game that could continue forever.

In array sizes of 2, you'd then have to check that the score doesn't exceed 10, and if it is equal to 10, then either the first throw is an 'X', or the second throw is a '/'.

A few ideas to put out there at least.